### PR TITLE
[WIP] Activitypub Service integrated into Backend

### PR DESCRIPTION
**Pull request by Mastercuber**
This integration is based on the `editor` branch of the source repository

### CODE REVIEW

**Comment by Lulalaby** [comment]
@Mastercuber don't forget to resolve the branch conflicts :D can't do it here.

**Comment by appinteractive** [comment]
@Mastercuber do that thing have to be inside the backend or could it be a standalone microservice?

**Comment by roschaefer** [comment]
@appinteractive I think there is no way around to touch the backend code if we want to get AcitivityPub compliance. An example would be if someone writes a post and has followers from Mastodon/Friendica. Only the backend knows about the new post and needs to send out messages.

Therefore I'm OK to incorporate another layer in our backend for activity pub. I suggested to merge @Mastercuber 's work as soon as possible, so it does not get out of sync.

File: package.json [hunk 1]
```diff
@@ -11,14 +11,16 @@
     "start": "node dist/",
     "dev": "nodemon --exec babel-node src/",
     "dev:debug": "nodemon --exec babel-node --inspect=0.0.0.0:9229 src/index.js",
-    "lint": "eslint src --config .eslintrc.js",
-    "test": "nyc  --reporter=text-lcov yarn run test:jest",
+    "lint": "eslint src --config .eslintrc.js --fix",
```
Comment (appinteractive) [inline package.json]: please do not run lint with fix as it is breaking the build pipleline and is annoying as hell, if you need it, run it menually with that parameter.
```suggestion
    "lint": "eslint src --config .eslintrc.js",
```

**Comment by appinteractive** [comment]
Somehow the branch status does not get updated... the conflicts are all resolved as you can see inside the changes... I don`t get it what is wrong here.

**Comment by appinteractive** [comment]
@Mastercuber in both varians (bevore merge and after) I have 3 failing cucumber tests:
grafik

File: package.json (hunk 1 above)
Comment (Mastercuber) [reply package.json]: Ok got it!

**Comment by Mastercuber** [comment]
@Lulalaby obviously Greg already removed the conflicts. Thanks!
@appinteractive It can also run as seperate service on an own port. The Nitro-Fediverse Repo is for this purpose now. 
For the tests not passing help is needed. When the test are manually "executed", you can see the changes through the graphql playground, but the tests fail. I think there is a problem with the asynchronouse execution of the tests.

**Comment by Mastercuber** [comment]
Locally all tests pass for me. Could someone please checkout the `activitypub-editor` branch and try running first the jest test `yarn test` and then the cucumber features with `yarn test:cucumber`
@roschaefer @Lulalaby @appinteractive @mattwr18.

**Comment by roschaefer** [comment]
@Mastercuber this is **awesome**, I just ran the tests and they pass locally. I'll have a look why the build server fails.

Excellent job :+1:

**Comment by roschaefer** [comment]
@Mastercuber I'll have to switch into another pair-programming, but could you re-run the tests with `docker-compose exec` (reproducing the build server)?

**Comment by Mastercuber** [comment]
@roschaefer Thanks =). I run the test like the build server is doing that. Locally they pass. Don't know what's wrong with the travis build..

**Comment by Mastercuber** [comment]
I restarted the build and obviously it's working, but don't know why!

File: README.md [hunk 2]
```diff
@@ -148,13 +148,20 @@ npm run db:reset
...
 npm run test
 ```
+Run the **_cucumber_** features:
+```bash
+yarn run test:cucumber
+# -or-
+npm run test:cucumber
+```
 
+When some tests fail, try `yarn db:reset` and after that `yarn db:seed`. Then run the tests again
```
Comment (roschaefer) [inline README.md]: :laughing:

**Comment by Mastercuber** [comment]
The test cases need to be extended and the updated activity is not covered now. 
Thank you for welcoming me :smiley:

File: README.md (hunk 2 above)
Comment (Mastercuber) [reply README.md]: Haha sometimes you need to do that for tests to pass ^^

**Comment by roschaefer** [comment]
Yay, tests have passed. @appinteractive this is a massive change of the backend code. I would consult you before merging. Considering you are currently ill I'm probably alone with the responsibility.

@Mastercuber I merge.

**Pull request merged**